pkg/core/tekton/tekton.go

```diff
@@ -0,0 +1,403 @@
+		}
+	}
+
+	prb.PipelineRunServiceAccount(pipelineRunServiceAccount)
+	prb.PipelineRunPipelineRef(p.Name)
+
+	prBytes, err := json.Marshal(prb.PipelineRun)
+	if err != nil {
+		log.Fatalf("Error marshalling PipelineRun: %s", err)
+	}
+	ttb.TriggerResourceTemplate(runtime.RawExtension{Raw: prBytes})
+
+	return &ttb.TriggerTemplate
+}
+
+func generateTriggerBinding(template *v1alpha1.TriggerTemplate) *v1alpha1.TriggerBinding {
+	webhookParamsMap := map[string]string{
+		codesetNameParam:    "$(body.repository.name)",
+		codesetVersionParam: "$(body.commits[0].id)",
+		codesetURLParam:     "$(body.repository.clone_url)",
+	}
+
+	tbb := builder.NewTriggerBindingBuilder(template.Name, template.Namespace)
+
+	for _, param := range template.Spec.Params {
+		if v, ok := webhookParamsMap[param.Name]; ok {
+			tbb.TriggerBindingParam(param.Name, v)
+		}
+	}
+	return &tbb.TriggerBinding
+}
+
+func generateEventListener(template *v1alpha1.TriggerTemplate, binding *v1alpha1.TriggerBinding) *v1alpha1.EventListener {
+	elb := builder.NewEventListenerBuilder(template.Name, template.Namespace)
+	elb.EventListenerServiceAccount(pipelineRunServiceAccount)
+	elb.EventListenerTriggerBinding(template.Name, binding.Name)
+	return &elb.EventListener
+}
+
+func toTektonTaskSpec(step workflow.WorkflowStep) v1beta1.TaskSpec {
+	tb := builder.NewTaskSpecBuilder(*step.Name, *step.Image, stepDefaultCmd)
+
+	for _, input := range step.Inputs {
+		// if there is a codeset as input, add workspace to the task and
+		// set its working directory to codeset.path
+		if input.Codeset != nil {
+			tb.TaskWorkspace(codesetWorkspaceName, *input.Codeset.Path)
+			tb.TaskWorkingDir(*input.Codeset.Path)
+		} else {
+			// else add it as a parameter to the tekton task
+			tb.TaskParam(*input.Name)
+			// make the inputs also available as env variable (with the
+			// FUSEML_ prefix) so that the container can use them
+			tb.TaskEnvVar(fmt.Sprintf("%s%s", inputsVarPrefix, strings.ToUpper(*input.Name)),
+				fmt.Sprintf("$(params.%s)", *input.Name))
+		}
+	}
+
+	// if image is parameterized reference it as a task parameter to be able
+	// to receive its value from a task output
+	if strings.Contains(*step.Image, "{{") {
+		tb.TaskParam(imageParamName, "Name (reference) of the image to run")
+		tb.TaskImage(fmt.Sprintf("$(params.%s)", imageParamName))
+	}
+
+	// a workflow output that is not an image represents a result in a tekton task.
+	// also adds an environment variable with the variable name so that the container
+	// can set the task output
+	for _, output := range step.Outputs {
+		if output.Image == nil {
+			tb.TaskResult(*output.Name)
+			tb.TaskEnvVar(stepOutputVarName, *output.Name)
+		}
+	}
+
+	// load environment variables
+	for _, stepEnv := range step.Env {
+		tb.TaskEnvVar(*stepEnv.Name, *stepEnv.Value)
+	}
+
+	return tb.TaskSpec
+}
+
+func stepOutputIsPipelineOutput(stepOutput *workflow.WorkflowStepOutput,
```

Comment:
I think this should be called `stepOutputIsWorkflowOutput` because it is operating over FuseML workflows, not over tekton pipelines

Reply:
indeed. thanks